File: app/src/main/java/com/github/marlonlom/demos/barcodescans/HomeActivity.java
package com.github.marlonlom.demos.barcodescans;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.TabLayout;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;

/**
 * Home activity class.
 *
 * @author marlonlom
 */
public class HomeActivity extends AppCompatActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        setupToolbar();
        setupTabs();
    }

    private void setupTabs() {
        final ViewPager viewPager = (ViewPager) findViewById(R.id.viewpager_main);
        setupViewPager(viewPager);

        TabLayout tabLayout = (TabLayout) findViewById(R.id.tablayout_main);
        tabLayout.setupWithViewPager(viewPager);
    }

    private void setupViewPager(final ViewPager viewPager) {
        final ViewPagerAdapter adapter = new ViewPagerAdapter(getSupportFragmentManager());
        adapter.addFragment(new DummyBarcodeScannerFragment());
        adapter.addFragment(new CameraBarcodeScannerFragment());
        viewPager.setAdapter(adapter);
    }

    private void setupToolbar() {
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar_main);
        setSupportActionBar(toolbar);
    }
}

File: app/src/main/java/com/github/marlonlom/demos/barcodescans/HomeActivity.kt
package com.github.marlonlom.demos.barcodescans

import android.os.Bundle
import android.support.design.widget.TabLayout
import android.support.v4.view.ViewPager
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar

/**
 * Home activity class.

 * @author marlonlom
 */
class HomeActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        setupToolbar()
        setupTabs()
    }

    private fun setupTabs() {
        val viewPager = findViewById(R.id.viewpager_main) as ViewPager
        setupViewPager(viewPager)

        val tabLayout = findViewById(R.id.tablayout_main) as TabLayout
        tabLayout.setupWithViewPager(viewPager)
    }

    private fun setupViewPager(viewPager: ViewPager) {
        val adapter = ViewPagerAdapter(supportFragmentManager)
        adapter.addFragment(DummyBarcodeScannerFragment())
        adapter.addFragment(CameraBarcodeScannerFragment())
        viewPager.adapter = adapter
    }

    private fun setupToolbar() {
        val toolbar = findViewById(R.id.toolbar_main) as Toolbar
        setSupportActionBar(toolbar)
    }
}
